### DISCUSSION TITLE
split --skip-deps in 2 flags

### DISCUSSION THREAD

**Discussion by jouve** [helmfile/helmfile#426] [Ideas]
```
      --skip-deps                    skip running "helm repo update" and "helm dependency build"
```

I would like to split this flag to mirror the behaviour of helm (build without refresh):
```
helm dep build -h
      --skip-refresh     do not refresh the local repository cache
```

to avoid many `helm repo up` which are quite slow on my setup.

**Comment by yxxhero** [comment]
@jouve I think it's a valuable idea.

@mumoshu WDYT?

**Comment by indrekj** [comment]
We'd like this as well. :+1:

**Comment by indrekj** [comment]
I tried `helmfile sync` with this patch:
```
diff --git a/pkg/helmexec/exec.go b/pkg/helmexec/exec.go
index 6c4d133..7a74e65 100644
--- a/pkg/helmexec/exec.go
+++ b/pkg/helmexec/exec.go
@@ -218,7 +218,7 @@ func (helm *execer) RegistryLogin(repository string, username string, password s
 
 func (helm *execer) BuildDeps(name, chart string) error {
        helm.logger.Infof("Building dependency release=%v, chart=%v", name, chart)
-       out, err := helm.exec([]string{"dependency", "build", chart}, map[string]string{}, nil)
+       out, err := helm.exec([]string{"dependency", "build", "--skip-refresh", chart}, map[string]string{}, nil)
        helm.info(out)
        return err
 }
```
and doing `helmfile repos` before running the `helmfile sync`.

The `sync` was almost 25% faster. We have about 40 releases and generally, this takes over 9 minutes. So this for us would be a significant boost. Note that it's not only `sync` and we're executing other things as well, so take these numbers with a grain of salt.

For us, it would be fine if `helmfile repos` is run before `runHelmDepBuilds` and then dependencies are run with `--skip-refresh` flag. In that case, it wouldn't even have to be a separate flag.

**Comment by yxxhero** [comment]
@indrekj PR is welcome. Thanks very much.

**Comment by indrekj** [comment]
https://github.com/helmfile/helmfile/pull/444
